rootfs/scheduler/__init__.py

```diff
@@ -65,35 +31,127 @@ def get_session():
 
 
 class KubeHTTPClient(object):
-    # used as the basis to check if a pod is ready
-    deploy_timeout = 120
-    apiversion = "v1"
+    # ISO-8601 which is used by kubernetes
+    DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
 
-    def __init__(self):
-        self.url = settings.SCHEDULER_URL
+    def __init__(self, url):
+        global resource_mapping
+        self.url = url
         self.session = get_session()
 
-    def log(self, namespace, message, level=logging.INFO):
+        # map the various k8s Resources to an internal property
+        from scheduler.resources import Resource  # lazy load
+        for res in Resource:
+            name = str(res.__name__).lower()  # singular
+            component = name + 's'  # make plural
+            # check if component has already been processed
+            if component in resource_mapping:
+                continue
+
+            # get past recursion problems in case of self reference
+            resource_mapping[component] = ''
+            resource_mapping[component] = res()
+            # map singular Resource name to the plural one
+            resource_mapping[name] = component
+            if res.short_name is not None:
+                # map short name to long name so a resource can be named rs
+                # but have the main object live at replicasets
+                resource_mapping[str(res.short_name).lower()] = component
+
+    def __getattr__(self, name):
+        global resource_mapping
+        if name in resource_mapping:
+            # resolve to final name if needed
+            component = resource_mapping[name]
+            if type(component) is not str:
+                # already a component object
+                return component
+
+            return resource_mapping[component]
+
+        return object.__getattribute__(self, name)
+
+    def version(self):
+        """Get Kubernetes version as a float"""
+        response = self.session.get(self.url + '/version')
+        if self.unhealthy(response.status_code):
+            raise KubeHTTPException(response, 'fetching Kubernetes version')
+
+        data = response.json()
+        return float('{}.{}'.format(data['major'], data['minor']))
+
+    @staticmethod
+    def parse_date(date):
+        return datetime.strptime(date, KubeHTTPClient.DATETIME_FORMAT)
+
+    @staticmethod
+    def unhealthy(status_code):
+        if not 200 <= status_code <= 299:
```

Comment:
Sure could, but it was just a copy of what we had there originally but brought into the class